beacon_node/eth2_libp2p/src/peer_manager/mod.rs

```diff
@@ -803,7 +794,7 @@ impl<TSpec: EthSpec> PeerManager<TSpec> {
     ) -> bool {
         {
             let mut peerdb = self.network_globals.peers.write();
-            if peerdb.is_banned(peer_id) {
+            if !matches!(peerdb.is_banned(peer_id), BanResult::NotBanned) {
```

Comment:
How come in the body of this `if` we don't return early? It seems we log a crit and then continue

Reply:
Yeah this is kind of a tricky situation. 
Libp2p manages our connections to peers, and we mimic those connections in the peer_manager. When we ban a peer, we tell libp2p to not accept any more connections. This function is where libp2p tells the peer manager we have established a new connection. 
If we have connected to a banned peer, there is something wrong with our banning logic. If we returned here, then the peer_manager would be out of sync with libp2p and we'd have extra connections we don't know about. So we wouldn't be able to disconnect them or even really know they are there. By not returning early, we still register the connection and the peer_manager should try and drop the peer again later or at least register its disconnection in some of the heartbeat cycles when we drop it for having a bad score.

Reply:
I.e the crit indicates a logic error in our code, but is entirely recoverable.

Reply:
Ah ok cool, that makes sense